### PR TITLE
Should add some first JavaDoc

### PR DESCRIPTION
**Pull request by duergner**
Not finalized yet - simply to get first feedback on documentation style. See #50 for further information / discussion.

@LappleApple / @jbellmann / @harti2006 can you have a look and let me know whether that style of documentation is fine for you?

@ePaul any feedback from your side whether this actually helps will also be welcome of course.

This PR should not yet be merged!

### CODE REVIEW

**Comment by LappleApple** [comment]
@duergner I'll proofread and edit after you merge. Currently I can't make any changes.

File: src/main/java/org/zalando/stups/tokens/AccessTokens.java [hunk 1]
```diff
@@ -15,11 +15,79 @@
...
+ * {@link AccessTokens#getAccessToken(Object)}.
+ *
+ * If you want to invalidate a specific access token to whatever reasons you can use
+ * {@link AccessTokens#invalidate(Object)}.
+ *
+ * The <i>tokenId</i> object used for all three methods must be an object that is equal to the
+ * object used when defining the token via {@link AccessTokensBuilder#manageToken(Object)}. If you
+ * are referring to tokens by {@link String} simply use a {@link String} with equals characters here.
+ *
+ * @author sarnowski
```
Comment (harti2006) [inline src/main/java/org/zalando/stups/tokens/AccessTokens.java]: I would omit the "@author" annotations, as they tend to get outdated very soon.

Comment (duergner) [inline src/main/java/org/zalando/stups/tokens/AccessTokens.java]: done

**Comment by harti2006** [comment]
:+1:

**Comment by jbellmann** [comment]
👍

File: src/main/java/org/zalando/stups/tokens/AccessTokens.java [hunk 2]
```diff
@@ -15,11 +15,74 @@
...
+ *
+ * The <i>tokenId</i> object used for all three methods must be an object that is equal to the
+ * object used when defining the token via {@link AccessTokensBuilder#manageToken(Object)}. If you
+ * are referring to tokens by {@link String} simply use a {@link String} with equals characters here.
+ */
 public interface AccessTokens {
+
+    /**
+     * Get a valid and up to date access token for the supplied <i>tokenId</i>. The returned {@link String}
+     * can be used as an OAuth2.0 Bearer token e.g.
```
Comment (ePaul) [inline src/main/java/org/zalando/stups/tokens/AccessTokens.java]: This sentence looks like not finished?

File: src/main/java/org/zalando/stups/tokens/AccessTokens.java [hunk 3]
```diff
@@ -15,11 +15,74 @@
...
+     * Get a full {@link AccessToken} not just the {@link String} representation for the supplied
+     * <i>tokenId</i>.
+     *
+     * @param tokenId  The <i>tokenId</i> to get the access token for. The supplied <i>tokenId</i>
+     *                 must be equal to the <i>tokenId</i> as supplied on
+     *                 {@link AccessTokensBuilder#manageToken(Object)} with respect to
+     *                 {@link Object#equals(Object)}.
+     * @return An {@link AccessToken} for the supplied <i>tokenId</i>. The return value will always
+     * be a non null value.
+     * @throws AccessTokenUnavailableException Thrown inn case there is no access token available
```
Comment (ePaul) [inline src/main/java/org/zalando/stups/tokens/AccessTokens.java]: inn → in

**Comment by ePaul** [comment]
Some minor typos, but in general it looks fine. Thank you.

**Pull request merged**